src/main/java/net/atos/zac/formulieren/FormulierDefinitieService.java

```diff
@@ -0,0 +1,89 @@
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Root;
+import javax.transaction.Transactional;
+
+import net.atos.zac.formulieren.model.FormulierDefinitie;
+
+@ApplicationScoped
+@Transactional
+public class FormulierDefinitieService {
+
+    @PersistenceContext(unitName = "ZaakafhandelcomponentPU")
+    private EntityManager entityManager;
+
+    public List<FormulierDefinitie> listFormulierDefinities() {
+        final CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+        final CriteriaQuery<FormulierDefinitie> query = builder.createQuery(FormulierDefinitie.class);
+        final Root<FormulierDefinitie> root = query.from(FormulierDefinitie.class);
+        query.orderBy(builder.asc(root.get("naam")));
+        query.select(root);
+        return entityManager.createQuery(query).getResultList();
+    }
+
+    public FormulierDefinitie readFormulierDefinitie(final long id) {
+        final FormulierDefinitie formulierDefinitie = entityManager.find(FormulierDefinitie.class, id);
+        if (formulierDefinitie != null) {
+            return formulierDefinitie;
+        } else {
+            throw new RuntimeException("%s with id=%d not found".formatted(FormulierDefinitie.class.getSimpleName(), id));
+        }
+    }
+
+    public FormulierDefinitie readFormulierDefinitie(final String systeemnaam) {
+        return findFormulierDefinitie(systeemnaam)
+                .orElseThrow(() -> new RuntimeException(
+                        "%s with code='%s' not found".formatted(FormulierDefinitie.class.getSimpleName(), systeemnaam)));
+    }
+
+    public Optional<FormulierDefinitie> findFormulierDefinitie(final String systeemnaam) {
+        final CriteriaBuilder builder = entityManager.getCriteriaBuilder();
+        final CriteriaQuery<FormulierDefinitie> query = builder.createQuery(FormulierDefinitie.class);
+        final Root<FormulierDefinitie> root = query.from(FormulierDefinitie.class);
+        query.select(root).where(builder.equal(root.get("systeemnaam"), systeemnaam));
+        final List<FormulierDefinitie> resultList = entityManager.createQuery(query).getResultList();
+        return resultList.isEmpty() ? Optional.empty() : Optional.of(resultList.get(0));
+    }
+
+    public FormulierDefinitie createFormulierDefinitie(final FormulierDefinitie formulierDefinitie) {
+        formulierDefinitie.setCreatiedatum(ZonedDateTime.now());
+        valideerObject(formulierDefinitie);
+        findFormulierDefinitie(formulierDefinitie.getSysteemnaam()).ifPresent(e -> {
+            if (!e.getId().equals(formulierDefinitie.getId())) {
+                throw new RuntimeException("Er bestaat al een formulier definitie met systeemnaam \"%s\"".formatted(formulierDefinitie.getSysteemnaam()));
```

Comment:
Ik zou hier een single quote gebruiken. Dat doen we in principe overal en doe jij ook in de RuntimeException op regel 52